src/h5web/visualizations/nexus/utils.ts

```diff
@@ -82,15 +83,13 @@ export function getNxDataGroup(
         ? getLinkedEntity(currSegment, parentEntity, metadata)
         : undefined;
     },
-    isAbsolutePath ? metadata.groups[metadata.root] : entity
+    isAbsolutePath ? metadata.groups[metadata.root] : group
   );
 
-  const nxDataGroup = getNxDataGroup(defaultEntity, metadata);
-  if (!nxDataGroup) {
-    throw new Error(`Expected to find NXdata group at path "${defaultPath}"`);
-  }
+  assertDefined(defaultEntity, `Expected entity at path "${defaultPath}"`);
+  assertGroup(defaultEntity, `Expected group at path "${defaultPath}"`);
```

Comment:
Now we can just assert that the entity found by the `reduce` is defined and is a group before recursing.